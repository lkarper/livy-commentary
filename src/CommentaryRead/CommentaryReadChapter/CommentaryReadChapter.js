import React from 'react';
import CommentaryReadSectionView from '../CommentaryReadSectionView/CommentaryReadSectionView';

const CommentaryReadChapter = (props) => {
   const { chapter } = props;

   return (
       <>
            <h3>Chapter {chapter.chapter_number}: {chapter.chapter_title}</h3>
            {chapter.intro_paragraphs.map((paragraph, i) => <p key={i}>{paragraph}</p>)}
            {chapter.sections.map(section => 
                    <CommentaryReadSectionView 
                        key={section.section_number} 
                        section={section} 
                    />)
            }
       </>
   );
}

export default CommentaryReadChapter;