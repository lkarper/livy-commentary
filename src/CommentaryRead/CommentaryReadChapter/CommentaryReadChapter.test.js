import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryReadChapter from './CommentaryReadChapter';

describe('CommentaryReadChapter component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <CommentaryReadChapter />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<CommentaryReadChapter />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
