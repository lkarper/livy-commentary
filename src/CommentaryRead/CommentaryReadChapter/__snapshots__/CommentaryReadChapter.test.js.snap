// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryReadChapter component renders the UI as expected 1`] = `
<CommentaryReadChapter
  chapter={
    Object {
      "chapter_intro": "",
      "chapter_number": "",
      "chapter_title": "",
      "sections": Array [],
    }
  }
>
  <h3>
    Chapter 
    : 
  </h3>
  <p />
</CommentaryReadChapter>
`;
